pkg/route/api/v1/types.go

```diff
@@ -43,6 +43,10 @@ type RouteSpec struct {
 	// be defaulted to Service.
 	To kapi.ObjectReference `json:"to"`
 
+	// AdditionalTos is an extension of the 'to' field. If more than one service needs to be
+	// pointed to, then use this field
+	AdditionalTos []kapi.ObjectReference `json:"additionalTos,omitempty"`
```

Comment:
That's a really, really, really obscure use case.  Maybe 1% of all users of
the router.  On the other hand, "how do I do a % A/B test" is going to be
much, much harder to use if you make them use annotations.



On Fri, May 27, 2016 at 6:56 PM, Rajat Chopra <notifications@github.com>
wrote:

> In pkg/route/api/v1/types.go
> <https://github.com/openshift/origin/pull/9070#discussion_r64975982>:
>
> > @@ -43,6 +43,10 @@ type RouteSpec struct {
> >  	// be defaulted to Service.
> >  	To kapi.ObjectReference `json:"to"`
> >
> > +	// AdditionalTos is an extension of the 'to' field. If more than one service needs to be
> > +	// pointed to, then use this field
> > +	AdditionalTos []kapi.ObjectReference `json:"additionalTos,omitempty"`
>
> "What's the point of AdditionalTos without weights? If we want to expose
> alternates, the only reason to do so is to weight them."
>
> The other use case brought about has been about source ip based switching.
> You may want a particular set of test-users belonging to an ip subnet to be
> able to access the alternate service. This is how it looks in haproxy:
>
> acl app1_privileged src 192.167.0.0/16
> acl app1_generic src 192.168.0.0/16
>
> use backend service1_hosts if app1_privileged
> use backend service2_hosts if app1_generic
>
> Same route, multiple services, nothing to do with weights.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9070/files/1da12ebe792ed12c5f100d4b8713be857a8232f7#r64975982>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p8Px-Z2cWccf0Vjdl-lX1N9Fzxcaks5qF3aRgaJpZM4Io5fr>
> .
>